lib/turn.rb

```diff
@@ -0,0 +1,68 @@
+class Turn
+    attr_reader :player1, :player2, :spoils_of_war
+    def initialize(player1, player2)
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+        @turn_type = nil
+    end
+
+    def type
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            @turn_type = :basic
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) && @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
+            @turn_type = :loss
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) && @player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)
+            @turn_type = :mutually_assured_destruction
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
+           @turn_type = :war        
+        end
+    end
+
+    def winner
+        if @turn_type == :mutually_assured_destruction
+            return "No Winner"
+        elsif @turn_type == :war
+            return @player1 if @player1.deck.rank_of_card_at(2) > @player2.deck.rank_of_card_at(2)
+            @player2
+        elsif @turn_type == :basic
+            return @player1 if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+            @player2 
+        elsif @turn_type == :loss
+            return @player1 if @player1.deck.cards.length > @player2.deck.cards.length
+            @player2
+        end
+    end
+
+    def pile_cards
+        if @turn_type == :mutually_assured_destruction
+            players = [@player1, @player2]
+            3.times do
+                players.each do |card|
+                    card.deck.cards.shift
```

Comment:
I vaguely remember there being a method on the `Deck` class that we defined that already does this `cards.shift` thing... 🤔 

```ruby
def remove_card
  cards.shift
end
```

Also, I think the block variable `|cards|` is not very descriptive of what it actually is. If we are iterating over the array of `players` defined on line 39, wouldn't each iteration be a `|player|` then?

Using `3.times` was a great way to avoid repetition!